DllImportGenerator/DllImportGenerator/TypePositionInfo.cs

```diff
@@ -31,56 +34,59 @@ internal sealed class TypePositionInfo
         private TypePositionInfo()
         {
             this.ManagedIndex = UnsetIndex;
-            this.UnmanagedIndex = UnsetIndex;
+            this.NativeIndex = UnsetIndex;
             this.UnmanagedLCIDConversionArgIndex = UnsetIndex;
         }
 
-        public ITypeSymbol TypeSymbol { get; private set; }
         public string InstanceIdentifier { get; private set; }
+        public ITypeSymbol ManagedType { get; private set; }
+        public ITypeSymbol NativeType { get; private set; }
```

Comment:
Do we need to have the `NativeType` field on `TypePositionInfo`? With the more complex marshallers, this info might not be quick to calculate. Additionally, if we are generating a P/Invoke call that takes a struct that has generated marshalling in the same source, we won't have an `ITypeSymbol` for the native type (since it's generated in a different source generator). Can we instead have the `NativeType` information be calculated and saved only in the respective marshaller generator objects (and all usage go through the `AsNativeType` method)?

Reply:
> we won't have an `ITypeSymbol` for the native type

Ahh, good call.